staging/src/k8s.io/api/resource/v1alpha2/types.go

```diff
@@ -18,6 +18,7 @@ package v1alpha2
 
 import (
 	v1 "k8s.io/api/core/v1"
+	namedresourcesv1alpha1 "k8s.io/api/resource/structured/namedresources/v1alpha1"
```

Comment:
From Slack (https://kubernetes.slack.com/archives/CHGFYJVAN/p1709327402062439?thread_ts=1709315837.565009&cid=CHGFYJVAN):

"let's not do this".

I'll move the `namedresources` types into the `resource` types.go files and add a `NamedResources` prefix to them.

Reply:
I've moved `resource/structured/namedresources/v1alpha1/types.go` into `resources/namedresources.go`, with `NamedResources` as prefix for the types. Still not "normal" and I could also copy into `types.go`, but I'd prefer to keep at least some separation if it's not causing problems. Apply configs get generated because the package has a `types.go`.

I have kept the `resource/structured` sub-directories for pure Go helper packages (`pkg/apis/resource/structured/namedresources/cel`, `pkg/apis/resource/structured/namedresources/validation`).

The downsides now are:
- `resource.NamedResourcesAllocationResult` instead of `namedresources.AllocationResult`, same for other types. Even worse is `resource.NamedResourcesResources` for node resources in NodeResourceSlice. Might need a better name. I'm leaving it for now, for the sake of consistency.
- No indication outside of comments that these types might be "alpha" in a future "beta" API.

cc @thockin